Turf/Turf.swift

```diff
@@ -107,126 +107,117 @@ public struct Turf {
-            let intersectionDistance: CLLocationDistance? = intersectionPoint != nil ? coordinate - intersectionPoint! : nil
-            
-            if distances.0 < closestCoordinate?.distance ?? CLLocationDistanceMax {
-                closestCoordinate = CoordinateAlongPolyline(coordinate: segment.0, index: index, distance: distances.0)
-            }
-            if distances.1 < closestCoordinate?.distance ?? CLLocationDistanceMax {
-                closestCoordinate = CoordinateAlongPolyline(coordinate: segment.1, index: index+1, distance: distances.1)
-            }
-            if intersectionDistance != nil && intersectionDistance! < closestCoordinate?.distance ?? CLLocationDistanceMax {
-                closestCoordinate = CoordinateAlongPolyline(coordinate: intersectionPoint!, index: (distances.0 < distances.1 ? index : index+1), distance: intersectionDistance!)
-            }
-        }
-        
-        return closestCoordinate
+    public static func wrap(_ value: Double, min minValue: Double, max maxValue: Double) -> Double {
+        let d = maxValue - minValue
+        return fmod((fmod((value - minValue), d) + d), d) + minValue
     }
     
     /**
-     Returns a subset of the polyline between given coordinates.
+     Returns the smallest angle between two angles.
      */
-    public static func polyline(along polyline: [CLLocationCoordinate2D], from start: CLLocationCoordinate2D? = nil, to end: CLLocationCoordinate2D? = nil) -> [CLLocationCoordinate2D] {
-        // Ported from https://github.com/Turfjs/turf/blob/142e137ce0c758e2825a260ab32b24db0aa19439/packages/turf-line-slice/index.js
-        guard !polyline.isEmpty else {
-            return []
-        }
-        
-        let startVertex = (start != nil ? closestCoordinate(on: polyline, to: start!) : nil) ?? CoordinateAlongPolyline(coordinate: polyline.first!, index: 0, distance: 0)
-        let endVertex = (end != nil ? closestCoordinate(on: polyline, to: end!) : nil) ?? CoordinateAlongPolyline(coordinate: polyline.last!, index: polyline.indices.last!, distance: 0)
-        let ends: (CoordinateAlongPolyline, CoordinateAlongPolyline)
-        if startVertex.index <= endVertex.index {
-            ends = (startVertex, endVertex)
-        } else {
-            ends = (endVertex, startVertex)
-        }
-        
-        var coords = ends.0.index == ends.1.index ? [] : Array(polyline[ends.0.index + 1...ends.1.index])
-        coords.insert(ends.0.coordinate, at: 0)
-        coords.append(ends.1.coordinate)
-        
-        return coords
+    public static func differenceBetweenAngles(_ alpha: CLLocationDirection, _ beta: CLLocationDirection) -> CLLocationDirection {
+        let phi = abs(beta - alpha).truncatingRemainder(dividingBy: 360)
+        return phi > 180 ? 360 - phi : phi
     }
+}
+
+public struct Polyline {
+    
+    public var coordinates: [CLLocationCoordinate2D]
```

Comment:
This property and the overall struct could use some short documentation comments.